lightning/src/ln/channelmanager.rs

```diff
@@ -7833,6 +7837,8 @@ where
 		let amount_msats = refund.amount_msats();
 		let relative_expiry = DEFAULT_RELATIVE_EXPIRY.as_secs() as u32;
 
+		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(self);
```

Comment:
Perhaps this just needs more clarification in the commit message, but why isn't this required in the `OffersMessageHandler` implementation? I might just not have a full grasp of the issue.

Reply:
Some what related: I was chatting with @tnull about adding an `InvoiceSent` event to help with LDK Node managing inbound payments. We create the invoice in the `OffersMessageHandler` implementation, but we don't send it until it goes through the `OnionMessenger`. By the time the `PeerManager` sends it, the invoice is no longer accessible as it's inside the `OnionMessage`. Wondering what the best strategy would be for generating such an event?
1. Generate it in `ChannelManager` before it is actually sent (thus needing persistence here)
2. Generate it in `OnionMessenger`, which we avoided for `ConnectionNeeded` (but event not persisted then)
3. Generate in `PeerManager` by pairing the `OnionMessage` with the (cloned) `Bolt12Invoice` payload (same, not persisted)

Seems like the options aren't that great. Any other alternatives?